martha/martha_v3.js

```diff
@@ -40,38 +147,70 @@ function marthaV3Handler(req, res) {
         res.status(BAD_REQUEST_ERROR_CODE).send(failureResponse);
         return;
     }
+    const parsedUrl = url.parse(dataObjectUri);
+    if (!parsedUrl.host || !parsedUrl.path) {
+        console.error(`"${url.format(parsedUrl)}" is missing a host and/or a path.`);
+        const failureResponse = new FailureResponse(BAD_REQUEST_ERROR_CODE, `"${dataObjectUri}" is not a properly-formatted URI.`);
+        res.status(BAD_REQUEST_ERROR_CODE).send(failureResponse);
+        return;
+    }
+    const {drsUrl, bondUrl, responseParser} = determineDrsType(parsedUrl);
+    console.log(`Converting DRS URI to HTTPS: ${dataObjectUri} -> ${drsUrl}`);
 
-    console.log(`Received URL '${dataObjectUri}' from IP '${req.ip}'`);
-    let dataObjectResolutionUrl;
+    let response;
     try {
-        dataObjectResolutionUrl = dataObjectUriToHttps(dataObjectUri);
+        response = await apiAdapter.getJsonFrom(drsUrl, auth);
```

Comment:
FYI- Going to submit a PR after this one to add a conditional around this line. We should only send the auth header to an allowed-list of hosts, not all. This line that always-sends-auth is a change from the previous `martha_v3` and `martha_v2` behavior.

Reply:
no need — I added a `sendAuth` property to the DrsType class